libs/core/execution_base/include/hpx/execution_base/traits/coroutine_traits.hpp

```diff
@@ -0,0 +1,227 @@
+//  Copyright (c) 2022 Shreyas Atre
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/config/forward.hpp>
+#include <hpx/concepts/has_member_xxx.hpp>
+
+#if defined(HPX_HAVE_CXX20_COROUTINES)
+
+#if __has_include(<coroutine>)
+#include <coroutine>
+namespace coro = std;
+#else
+#include <experimental/coroutine>
+namespace coro = std::experimental;
+#endif
```

Comment:
I'd suggest aliasing the namespaces to `namespace hpx::coro` to avoid name clashes for users. E.g. something like:
```suggestion
#if __has_include(<coroutine>)
#include <coroutine>
namespace hpx {
    namespace coro = std;
}
#else
#include <experimental/coroutine>
namespace hpx {
    namespace coro = std::experimental;
}
#endif
```
or even better import the necessary names into `hpx::coro` instead of aliasing the whole namespace `std`. This should also go into a separate header file to avoid duplicating the declarations

Reply:
Thanks :+1: